platform/wayland/cog-platform-wl.c

```diff
@@ -1968,24 +2188,18 @@ init_wayland (GError **error)
 }
 
 static void
-clear_wayland (void)
+clear_wayland(void)
 {
+    win_data.shell_context->functions.destroy_shell();
```

Comment:
Just echoing my previous comment on this, I think it would be good to clear the shell_context pointer after calling destroy_shell.